src/bundle.rs

```diff
@@ -13,6 +13,75 @@ use core::{fmt, mem};
 use crate::archetype::TypeInfo;
 use crate::Component;
 
+/// Checks if a query is satisfied by a bundle. This is primarily useful for unit tests.
+///
+/// ```rs
+/// #[derive(hecs::Bundle)]
+/// struct MyBundle {
+///     health: i32,
+///     speed: f32,
+/// }
+///
+/// #[derive(hecs::Query)]
+/// struct MyQuery<'a> {
+///     health: &'a i32,
+///     speed: &'a f32,
+/// }
+///
+/// #[cfg(test)]
+/// mod tests {
+///     use super::{MyBundle, MyQuery};
+///     
+///     #[test]
+///     fn my_bundle_satisfies_my_query() {
```

Comment:
For concision, I think we can omit all this boilerplate. Just the bare assert after the definitions should be fine.